packages/react/src/components/Tooltip/Tooltip.tsx

```diff
@@ -0,0 +1,290 @@
+    mountNode: isBrowser() ? document.body : null,
+    position: 'above',
+    mouseLeaveDelay: 500,
+    pointing: true,
+  }
+
+  static autoControlledProps = ['open']
+
+  pointerTargetRef = React.createRef<HTMLElement>()
+  triggerRef = React.createRef<HTMLElement>() as React.MutableRefObject<HTMLElement>
+  tooltipDomElement = null
+
+  closeTimeoutId
+
+  renderComponent({
+    classes,
+    rtl,
+    accessibility,
+  }: RenderResultConfig<TooltipProps>): React.ReactNode {
+    const { mountNode } = this.props
+    const tooltipContent = this.renderTooltipContent(classes.content, rtl, accessibility)
+
+    return (
+      <>
+        {this.renderTrigger(accessibility)}
+        {mountNode && ReactDOM.createPortal(tooltipContent, mountNode)}
+      </>
+    )
+  }
+
+  getTriggerProps(triggerElement) {
+    const triggerProps: any = {}
+
+    triggerProps.onFocus = (e, ...args) => {
+      if (isFromKeyboard()) {
+        this.trySetOpen(true, e)
+      }
+      _.invoke(triggerElement, 'props.onFocus', e, ...args)
+    }
+    triggerProps.onBlur = (e, ...args) => {
+      if (this.shouldBlurClose(e)) {
+        this.trySetOpen(false, e)
+      }
+      _.invoke(triggerElement, 'props.onBlur', e, ...args)
+    }
+
+    triggerProps.onMouseEnter = (e, ...args) => {
+      this.setTooltipOpen(true, e)
+      setWhatInputSource('mouse')
+      _.invoke(triggerElement, 'props.onMouseEnter', e, ...args)
+    }
+    triggerProps.onMouseLeave = (e, ...args) => {
+      this.setTooltipOpen(false, e)
+      _.invoke(triggerElement, 'props.onMouseLeave', e, ...args)
+    }
+
+    return triggerProps
+  }
+
+  getContentProps = (predefinedProps?) => {
+    const contentHandlerProps: any = {}
+
+    contentHandlerProps.onMouseEnter = (e, contentProps) => {
+      this.setTooltipOpen(true, e)
+      predefinedProps && _.invoke(predefinedProps, 'onMouseEnter', e, contentProps)
+    }
+    contentHandlerProps.onMouseLeave = (e, contentProps) => {
+      this.setTooltipOpen(false, e)
+      predefinedProps && _.invoke(predefinedProps, 'onMouseLeave', e, contentProps)
+    }
+
+    return contentHandlerProps
+  }
+
+  shouldBlurClose = e => {
+    return (
+      !e.currentTarget ||
+      !this.tooltipDomElement ||
+      (!e.currentTarget.contains(e.relatedTarget) &&
+        !this.tooltipDomElement.contains(e.relatedTarget))
+    )
+  }
+
+  renderTrigger(accessibility) {
+    const { children, trigger } = this.props
+    const triggerElement = childrenExist(children) ? children : (trigger as any)
+    const triggerProps = this.getTriggerProps(triggerElement)
+    return (
+      triggerElement && (
+        <Ref innerRef={this.triggerRef}>
+          {React.cloneElement(triggerElement, {
+            ...accessibility.attributes.trigger,
+            ...triggerProps,
+            ...applyAccessibilityKeyHandlers(accessibility.keyHandlers.trigger, triggerProps),
+          })}
+        </Ref>
+      )
+    )
+  }
+
+  renderTooltipContent(
```

Comment:
`renderTooltipContent` and `renderPopperChildren` have a bit of duplicated code with `Popup.tsx` - anything worth reusing? treat this comment as NIT